src/main/java/frc/team2412/robot/Robot.java

```diff
@@ -168,9 +157,9 @@ public void startCompetition() {
 
     @Override
     public void robotInit() {
-        hardware = new Hardware();
-        subsystems = new Subsystems(hardware);
-        controls = new Controls(subsystems);
+        hardware = hardware.instance;
+        subsystems = subsystems.instance;
+        controls = controls.instance;
```

Comment:
I'd recommend reading on "singleton anti-pattern" for why this is something that initially seems good, but in practice ends badly. Please undo the conversion of things to singletons.